File: Sources/IssueCaptureUnity/UnityCommand.swift
import Foundation
import IssueCapture

struct UnityCommand: Decodable {
    let operation: String
    let session: String
    var projectID: String?
    var sourceRevision: String?
    var eventLimit: Int?
    var eventByteLimit: Int?
    var screen: UnityScreen?
    var active: Bool?
    var category: String?
    var name: String?
    var result: String?
    var file: String?
    var line: UInt?
}

struct UnityScreen: Decodable {
    let id: UUID
    let parentID: String
    let stableID: String
    let name: String
    let typeName: String
    let file: String
    let line: UInt

    var context: IssueScreenContext {
        .init(id: id, stableID: String(stableID.prefix(200)), name: String(name.prefix(200)),
              typeName: String(typeName.prefix(300)), file: String(file.prefix(300)),
              line: line, parentID: UUID(uuidString: parentID))
    }
}

File: Unity/Runtime/Model/UnityCommand.cs
using System;

namespace VinWare.IssueCapture
{
    [Serializable]
    internal sealed class ScreenContext
    {
        public string id, parentID, stableID, name, typeName, file;
        public int line;
    }

    [Serializable]
    internal sealed class Command
    {
        public string operation, session, category, name, result, file;
        public string projectID = null, sourceRevision = null;
        public int eventLimit = 50, eventByteLimit = 262144, line;
        public bool active;
        public ScreenContext screen;
    }
}
